sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -92,67 +94,282 @@ public HttpPipeline getHttpPipeline() {
         return withContext(context -> createDigitalTwinWithResponse(digitalTwinId, digitalTwin, context));
     }
 
+    Mono<DigitalTwinsResponse<String>> createDigitalTwinWithResponse(String digitalTwinId, String digitalTwin, Context context) {
+        return protocolLayer
+            .getDigitalTwins()
+            .addWithResponseAsync(digitalTwinId, digitalTwin, context)
+            .flatMap(response -> {
+                try {
```

Comment:
We want to convert the item emitted by `addWithResponseAsync(digitalTwinId, digitalTwin, context)` into json string representation, and return it in a `Mono`; or return a `Mono.error` if the serialization fails.
So we need to use the `Mono.flatMap` API: https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html#flatMap-java.util.function.Function-

Reply:
`flatMap(Function<? super T,? extends Mono<? extends R>> transformer)`
Transform the item emitted by this Mono asynchronously, returning the value emitted by another Mono (possibly changing the value type).

Reply:
Thank you for explanation.